### PR TITLE
Message sending

### PR DESCRIPTION
**Pull request by schuyler1d**
addresses https://github.com/MoveOnOrg/connier/issues/99

semi-WIP:  this allows bulk message sending and sending on a generic object
Still TODO: 
1. access model, 
2. saving messages, 
3. cleanup in event_review/admin.py for only one message interface

Even though there are other parts to do, I think it would be worth getting this reviewed and merging it into election2018 -- so the connier parts can get added, and cleanup can be done without interfering with other folks' commits/work.

### CODE REVIEW

File: reviewer/message_sending.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return urls + my_urls
+
+    def send_message(self, request, organization, obj_id):
+        result = 'failed to send'
+        if not ReviewGroup.user_allowed(request.user, organization):
+            result = 'permission denied'
+        elif request.method == 'POST' and self.message_send_ready():
+            obj = self.message_obj_lookup(obj_id, organization, request)
+            if obj:
+                self.send_messages(request.POST.get('message',''), [event])
```
Comment (schuyler1d) [inline reviewer/message_sending.py]: `[event]` should be `[obj]`

File: reviewer/templates/reviewer/admin/bulk_message_send.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    <script type="text/javascript" src="{% static 'admin/js/cancel.js' %}"></script>
+    <link rel="stylesheet" type="text/css" href="/static/css/site.css" />
+
+    <link rel="apple-touch-icon" href="//s3.amazonaws.com/s3.moveon.org/images/with_dims/moveon_iphone_icon_57x57.png"/>
+    <link rel="shortcut icon" href="//s3.amazonaws.com/s3.moveon.org/favicon.ico" />
+    <link href="//s3.amazonaws.com/actionkit.moveon.org/static/stylesheets/site.css" rel="stylesheet">
+    <link href="//netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap-glyphicons.css" rel="stylesheet">
+{% endblock %}
+
+{% block bodyclass %}{{ block.super }} app-{{ opts.app_label }} model-{{ opts.model_name }} delete-confirmation delete-selected-confirmation{% endblock %}
```
Comment (sandramchung) [inline reviewer/templates/reviewer/admin/bulk_message_send.html]: This template looks familiar! it probably doesn't matter, but some of these classnames might be confusing in this context.

Comment (schuyler1d) [reply reviewer/templates/reviewer/admin/bulk_message_send.html]: Yes -- I copied yours!  thank you!  I removed the incriminating evidence^H^H^H^H^H^H extraneous classes :-)

File: event_review/admin.py [hunk 3]
```diff
@@ -200,3 +221,41 @@ def get_queryset(self, *args, **kw):
...
         qs = qs.select_related('organization_host', 'organization_source', 'organization')
         return qs
+
+    ## BEGIN EventAdmin Message Sending
+    send_a_message_placeholder = 'Optional message to host. Email will include a link to manage the event.'
+    def message_template(self, message, event):
+        """
+        NOTE: This takes a while to render, almost entirely because
+          get_host_event_link needs to get a login token by AK API, which takes some time.
+          This will SLOW mass message delivery
```
Comment (mathemagica) [inline event_review/admin.py]: should we consider bulk requesting these ahead of time? like maybe in a scheduled job that requests tokens for all current hosts of events in whatever we say the active event campaign is

**Pull request merged**